kernel/safe_typing.ml

```diff
@@ -336,7 +338,7 @@ let push_context_set poly cst senv =
   if Univ.ContextSet.is_empty cst then senv
   else
     let sections =
-      if Section.is_empty senv.sections then senv.sections
+      if not (sections_are_opened senv) then senv.sections
```

Comment:
This is doing a bit of double-negation logic for free. Why didn't you like the `is_empty` primitive?

Reply:
I dunno, I can revert that bit if you prefer

Reply:
I'm not strongly advocating for it, but the code would be cleaner and the PR touching less lines of code, so...